File: 2026_GFG_POTD/JANUARY/27-01-2026/Word Search.cs
class Solution {
    int n, m;

    bool Dfs(int i, int j, int idx, char[,] mat, string word) {
        if (idx == word.Length) return true;
        if (i < 0 || j < 0 || i >= n || j >= m || mat[i, j] != word[idx]) return false;

        char temp = mat[i, j];
        mat[i, j] = '#';

        bool found = Dfs(i + 1, j, idx + 1, mat, word) ||
                     Dfs(i - 1, j, idx + 1, mat, word) ||
                     Dfs(i, j + 1, idx + 1, mat, word) ||
                     Dfs(i, j - 1, idx + 1, mat, word);

        mat[i, j] = temp;
        return found;
    }

    public bool isWordExist(char[,] mat, string word) {
        n = mat.GetLength(0);
        m = mat.GetLength(1);

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (Dfs(i, j, 0, mat, word)) return true;
            }
        }
        return false;
    }
}

File: 2026_GFG_POTD/JANUARY/27-01-2026/Word Search.js
class Solution {
    isWordExist(mat, word) {
        const n = mat.length, m = mat[0].length;

        const dfs = (i, j, idx) => {
            if (idx === word.length) return true;
            if (i < 0 || j < 0 || i >= n || j >= m || mat[i][j] !== word[idx]) return false;

            const temp = mat[i][j];
            mat[i][j] = '#';

            const found = dfs(i + 1, j, idx + 1) ||
                          dfs(i - 1, j, idx + 1) ||
                          dfs(i, j + 1, idx + 1) ||
                          dfs(i, j - 1, idx + 1);

            mat[i][j] = temp;
            return found;
        };

        for (let i = 0; i < n; i++) {
            for (let j = 0; j < m; j++) {
                if (dfs(i, j, 0)) return true;
            }
        }
        return false;
    }
}